drivers/can/stm32_can.c

```diff
@@ -215,8 +215,11 @@ int can_stm32_runtime_configure(struct device *dev, enum can_mode mode,
 	__ASSERT_NO_MSG(clock);
 	hcan.Instance = can;
 
-	clock_control_get_rate(clock, (clock_control_subsys_t *) &cfg->pclken,
-			       &clock_rate);
+	if (clock_control_get_rate(clock, (clock_control_subsys_t *) &cfg->pclken,
```

Comment:
please call function outside of the if statement and check return value instead.